tests/main/snap-user-service-socket-activation/task.yaml

```diff
@@ -0,0 +1,47 @@
+summary: Check that sockets for user services are created correctly
+
+systems:
+    # Ubuntu 14.04's systemd doesn't have user@.service
+    - -ubuntu-14.04-*
+    # Amazon Linux 2 gives error "Unit user@12345.service not loaded."
+    - -amazon-linux-2-*
+    # Centos 7 gives error "Unit user@12345.service not loaded."
+    - -centos-7-*
+
+kill-timeout: 5m
+
+prepare: |
+    snap set system experimental.user-daemons=true
+    session-tool --kill-leaked
+
+restore: |
+    session-tool -u test --restore
+    snap unset system experimental.user-daemons
+
+execute: |
+    echo "When the service snap is installed"
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    install_local test-snapd-user-service-sockets
+
+    echo "And the user mode systemd instance is started"
+    session-tool -u test --prepare
+
+    echo "And the user session has finished starting"
```

Comment:
Those two can now go away

Reply:
Removing these caused a failure on the Ubuntu 16.04 based systems (both classic and core).

From the look of it, `session-tool --prepare` will only wait for `default.target` if there is a systemd controlled session bus.  Ubuntu 16.04's session bus is managed by Upstart, yet it has a functioning user mode systemd instance.

There's no dependency on the D-Bus session bus for many operations, even with more recent systemd.  For instance, on 20.04:

```
$ strace --trace=connect systemctl --user is-active snapd.session-agent.service
connect(3, {sa_family=AF_UNIX, sun_path="/run/user/1000/systemd/private"}, 33) = 0
inactive
+++ exited with 3 +++
```

It is communicating with systemd over a private socket.